components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutionResponseProcessorFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.execution;
+
+import org.wso2.carbon.identity.action.execution.model.ActionType;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class defines the Action Execution Response Processor Factory.
+ * Action Execution Response Processor Factory is the component that is responsible for providing the
+ * {@link ActionExecutionResponseProcessor} based on the action type.
+ */
+public class ActionExecutionResponseProcessorFactory {
+
+    private static final Map<ActionType, ActionExecutionResponseProcessor> actionInvocationResponseProcessors =
+            new HashMap<>();
+
+    public static ActionExecutionResponseProcessor getActionExecutionResponseProcessor(ActionType actionType) {
+
+        switch (actionType) {
```

Comment:
Shall we use an if block here?

Reply:
We will be onboarding multiple action types. In this context we don't need it but later we do. So let's just keep it and move forward.